go-controller/pkg/ovn/controller/services/loadbalancer.go

```diff
@@ -18,6 +18,44 @@ import (
 	"k8s.io/klog/v2"
 )
 
+const LBVipNodeTemplate string = "NODEIP"
+
+// makeLBNodeIPTemplateName creates a template name for the node IP (per family)
+func makeLBNodeIPTemplateName(family corev1.IPFamily) string {
+	return fmt.Sprintf("%s_%v", LBVipNodeTemplate, family)
+}
+
+// isLBNodeIPTemplateName returns true if 'name' is the node IP template name
+// for any IP family.
+func isLBNodeIPTemplateName(name string) bool {
+	return name == makeLBNodeIPTemplateName(corev1.IPv4Protocol) ||
+		name == makeLBNodeIPTemplateName(corev1.IPv6Protocol)
+}
+
+// makeLBTargetTemplateName builds a load balancer target template name.
+func makeLBTargetTemplateName(service *corev1.Service, proto corev1.Protocol, port int32,
+	family corev1.IPFamily, scope string) string {
+	return makeTemplateName(
+		makeLBName(service, proto,
+			fmt.Sprintf("%d_%s_%v", port, scope, family)))
+}
+
+// getTemplatesFromRules returns the map of template variables referred to by 'rules'.
+func getTemplatesFromRules(rules []LBRule) TemplateMap {
+	templates := TemplateMap{}
+	for _, r := range rules {
+		for _, tgt := range r.Targets {
+			if tgt.Template == nil || tgt.Template.Name == "" {
+				continue
+			}
+			templates[tgt.Template.Name] = tgt.Template
+		}
+		// No need to return source templates, those are managed in the
+		// node-tracker.
+	}
+	return templates
+}
```

Comment:
nit:  I think these should potentially go into `svc_template_var.go`

Reply:
Ok, moved.